templates/taskcat-ci-entrypoint.template.yaml

```diff
@@ -0,0 +1,506 @@
+
+          phases:
+            install:
+              runtime-versions:
+                python: 3.x
+              commands:
+                - echo "Entered the install phase..."
+                - apt-get -qy update
+                - echo "Installing system dependencies..."
+                - apt-get -qy install zip gzip tar
+                - echo "Installing python dependencies..."
+                - pip3 install cfn-lint taskcat yq
+                - echo "Configuring taskcat..."
+                - |
+                  cat << EOF > ~/.taskcat.yml
+                  general:
+                    s3_regional_buckets: true
+                  EOF
+                - echo "Installing custom cfn-lint rules..."
+                - wget -nv https://github.com/aws-quickstart/qs-cfn-lint-rules/archive/refs/heads/main.zip
+                - unzip -q ./main.zip
+                - mv ./qs-cfn-lint-rules-main ~/qs-cfn-lint-rules
+                - pip3 install -e ~/qs-cfn-lint-rules
+                - |
+                  cat << EOF > ~/.cfnlintrc
+                  append_rules:
+                    - ~/qs-cfn-lint-rules/qs_cfn_lint_rules/
+                  EOF
+            pre_build:
+              commands:
+                - echo "Entered the pre_build phase..."
+                - echo "Current directory is $CODEBUILD_SRC_DIR"
+                - ls -lA
+                - dirname=${!PWD##*/}
+                - echo "Directory name $dirname"
+                - cd ..
+                - mv $dirname $PROJECT_NAME
+                - ls -lA
+                - cd $PROJECT_NAME
+                - echo "https://$GITHUB_USERNAME:$GITHUB_TOKEN@github.com" > ~/.git-credentials
+                - git config --global credential.helper store
+                - git config --global url."https://github.com/".insteadOf "git@github.com:"
+                - git config --global init.defaultBranch ${ReleaseBranch}
+                - git init
+                - git remote add origin https://github.com/$PROJECT_OWNER/$PROJECT_NAME.git
+                - git fetch
+                - git checkout --force --track origin/$SOURCE_BRANCH
+                - |
+                  git config -f .gitmodules --get-regexp '^submodule\..*\.path$' |
+                    while read path_key path
+                    do
+                        url_key=$(echo $path_key | sed 's/\.path/.url/')
+                        url=$(git config -f .gitmodules --get "$url_key")
+                        git submodule add $url $path
+                    done
+                - git submodule init
+                - git submodule update --init --recursive
+                - echo "Creating TaskCat overrides file..."
+                - echo $TASKCAT_OVERRIDES | yq -y > ./.taskcat_overrides.yml
+                - ls -lA
+                - echo "Verifying TaskCat installation..."
+                - taskcat
+            build:
+              commands:
+                - echo "Entered the build phase..."
```

Comment:
Somewhere before here, please remove the `~/.gitcredentials` file; It shouldn't be uplaoded, but out of an abundance of caution....

Reply:
Good call. Added in 41bb614a85c8b3a9488364d895197ffe0cb436cb.